src/components/GraphiQL/GraphiQL.tsx

```diff
@@ -0,0 +1,346 @@
+      "The `GraphiQL` component requires a `fetcher` function to be passed as prop.",
+    );
+  }
+
+  return (
+    <GraphiQLProvider
+      getDefaultFieldNames={getDefaultFieldNames}
+      dangerouslyAssumeSchemaIsValid={dangerouslyAssumeSchemaIsValid}
+      defaultQuery={defaultQuery}
+      defaultHeaders={defaultHeaders}
+      defaultTabs={defaultTabs}
+      externalFragments={externalFragments}
+      fetcher={fetcher}
+      headers={headers}
+      initialTabs={initialTabs}
+      inputValueDeprecation={inputValueDeprecation}
+      introspectionQueryName={introspectionQueryName}
+      maxHistoryLength={maxHistoryLength}
+      onEditOperationName={onEditOperationName}
+      onSchemaChange={onSchemaChange}
+      onTabChange={onTabChange}
+      onTogglePluginVisibility={onTogglePluginVisibility}
+      plugins={plugins}
+      visiblePlugin={visiblePlugin}
+      operationName={operationName}
+      query={query}
+      response={response}
+      schema={schema}
+      schemaDescription={schemaDescription}
+      shouldPersistHeaders={shouldPersistHeaders}
+      storage={storage}
+      validationRules={validationRules}
+      variables={variables}
+    >
+      <GraphiQLInterface {...props} />
+    </GraphiQLProvider>
+  );
+}
+// Export main windows/panes to be used separately if desired.
+GraphiQL.Toolbar = GraphiQLToolbar;
+
+type AddSuffix<Obj extends Record<string, any>, Suffix extends string> = {
+  // @ts-ignore
+  [Key in keyof Obj as `${string & Key}${Suffix}`]: Obj[Key];
+};
+
+export type GraphiQLInterfaceProps = WriteableEditorProps &
+  AddSuffix<Pick<UseQueryEditorArgs, "onEdit">, "Query"> &
+  Pick<UseQueryEditorArgs, "onCopyQuery"> &
+  AddSuffix<Pick<UseVariableEditorArgs, "onEdit">, "Variables"> &
+  AddSuffix<Pick<UseHeaderEditorArgs, "onEdit">, "Headers"> &
+  Pick<UseResponseEditorArgs, "responseTooltip"> & {
+    children?: ReactNode;
+    defaultEditorToolsVisibility?: boolean | "variables" | "headers";
+    isHeadersEditorEnabled?: boolean;
+    toolbar?: GraphiQLToolbarConfig;
+  };
+
+export function GraphiQLInterface(props: GraphiQLInterfaceProps) {
+  const editorContext = useEditorContext({ nonNull: true });
+  const pluginContext = usePluginContext();
+
+  const classes = useStyles({});
+
+  const copy = useCopyQuery({ onCopyQuery: props.onCopyQuery });
+  const prettify = usePrettifyEditors();
+
+  const PluginContent = pluginContext?.visiblePlugin?.content;
+
+  const pluginResize = useDragResize({
+    defaultSizeRelation: 1 / 3,
+    direction: "horizontal",
+    initiallyHidden: pluginContext?.visiblePlugin ? undefined : "first",
+    sizeThresholdSecond: 200,
+    storageKey: "docExplorerFlex",
+  });
+  const editorResize = useDragResize({
+    direction: "horizontal",
+    storageKey: "editorFlex",
+  });
+  const editorToolsResize = useDragResize({
+    defaultSizeRelation: 3,
+    direction: "vertical",
+    sizeThresholdSecond: 60,
+    storageKey: "secondaryEditorFlex",
+  });
```

Comment:
these hooks are related to the styling context, so:

**suggestion**: how about encapsulating them within a separate hook? then we have one responsibility there `const { pluginResize, editorResize, toolsResize} = useResize(pluginContext) // or useEditorStyles()`

Reply:
@andrzejewsky Done. For the context, this code is directly taken from the GraphiQL implementation. The intention was to leave it as-is to facilitate the potential updates that happen in that package upstream.